esprima.js

```diff
@@ -4369,12 +4478,20 @@
                 token = lookahead;
                 isStatic = false;
                 computed = match('[');
-                key = parseObjectPropertyKey();
-                if (key.name === 'static' && lookaheadPropertyName()) {
+                if (match('*')) {
+                    lex();
+                } else {
+                    key = parseObjectPropertyKey();
+                }
+                if (key && key.name === 'static' && (lookaheadPropertyName() || match('*'))) {
```

Comment:
I think you can just move this `if` into the body of the immediately previous `else` and forgo the `key &&` here